### DISCUSSION TITLE
Lesson 11 - Interactions.s.sol

### DISCUSSION THREAD

**Discussion by 0xSimeon** [Cyfrin/foundry-full-course-cu#400] [Q&A] [answered]
I know this has been done a couple of times during the course but I'm wondering why this works. 

Full code: 
```solidity
// SPDX-License-Identifier: MIT

pragma solidity ^0.8.18;

import {Script} from "forge-std/Script.sol";
import {DevOpsTools} from "lib/foundry-devops/src/DevOpsTools.sol";
import {BasicNft} from "../src/BasicNft.sol";

contract MintBasicNft is Script {
    string public constant PUG =
        "ipfs://bafybeig37ioir76s7mg5oobetncojcm3c3hxasyd4rvid4jqhy4gkaheg4/?filename=0-PUG.json";

    function run() external {
        address mostRecentlyDeployed = DevOpsTools.get_most_recent_deployment(
            "BasicNft", block.chainid
        );

        mintNftOnContract(mostRecentlyDeployed);
    }

    function mintNftOnContract(address contractAddress) public {
        vm.startBroadcast();
        BasicNft(contractAddress).mintNft(PUG);
        vm.stopBroadcast();

    }
}
```

The line I'm confused about is `BasicNft(contractAddress).mintNft(PUG);`. Our **BasicNft** contract doesn't have constructor params so what is `BasicNft(contractAddress)` actually doing? I would appreciate if someone can help me out with an explanation, Thank you 🙂.

**Comment by ghost** [comment, marked as answer]
Hello @0xSimeon.

`BasicNft(contractAddress)` doesn't deploy a new contract it casts or wraps an address to contract creating a pointer to a contract that has the structure of `BasicNft`.

After the pointer is created we can interact with the methods in the contract, in this case a call to `mintNft` with its param. This is similar to how you can access the methods from a class in JavaScript.
